src/nix/profile.cc

```diff
@@ -453,55 +453,117 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
     }
 };
 
-class MixProfileElementMatchers : virtual Args
+struct Matcher
 {
-    std::vector<std::string> _matchers;
+    std::string name;
+    std::function<bool(const std::string & name, const ProfileElement & element)> matches;
+};
+
+Matcher createRegexMatcher(const std::string & pattern)
+{
+    std::regex reg(pattern, std::regex::extended | std::regex::icase);
+    return {
+        .name = fmt("Regex '%s'", pattern),
+        .matches = [reg](const std::string &name, const ProfileElement & element) {
+            return std::regex_match(element.identifier(), reg);
+        },
+    };
+}
+
+Matcher createStorePathMatcher(const nix::StorePath & storePath)
+{
+    return {
+        .name = fmt("Store path '%s'", storePath.to_string()),
+        .matches = [storePath](const std::string &name, const ProfileElement & element) {
+            return element.storePaths.count(storePath);
+        }
+    };
+}
+
+Matcher createNameMatcher(const std::string & name) {
+    return {
+        .name = fmt("Package name '%s'", name),
+        .matches = [name](const std::string &elementName, const ProfileElement & element) {
+            return elementName == name;
+        }
+    };
+}
+
+Matcher all = {
+    .name = "All",
```

Comment:
```suggestion
    .name = "--all",
```
Otherwise you might get an error like "All does not match any element...".

Reply:
Yes, that's also what I ran into. That scenario would only happen when there are no elements in the profile, but I've explicitly made an error for that case to avoid such a generic error :+1:

Might still be good to make the change for future use in error/warnings.